### PR TITLE
Add Car and anothter *.java

### PR DESCRIPTION
**Pull request by Andrij72**

### CODE REVIEW

**Comment by Andrij72** [comment]
Small different changers in Car.java

File: Car.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
+package com.mateakademia.homework.homework3.task32.lamborghinicar;
+
+import java.time.LocalDate;
+
+public class Car {
+    private final int produceDate;
```
Comment (Anderwerty) [inline Car.java]: Please use type LocalDate for productDate.

File: Car.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    public void getOffAllCarWheels(){
+        for (CarWheel carwheel: carwheels){
+            carwheel = null;
+        }
+        wheelCount = 0;
+    }
+
+    public void setNewWheels(int x){
```
Comment (Anderwerty) [inline Car.java]: Please, use a better name for the parameter. Example: x->numberOfWheels

File: Car.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        } else {
+            for (int i = 0; i < x; i++){
+                carwheels[wheelCount + i] = new CarWheel();
+            }
+            wheelCount += x;
+        }
+    }
+
+    public double getCurrentMaxSpeed(double minTWheel){
+        if ((currentPassNumber == 0) || (wheelCount == 0)){
```
Comment (Anderwerty) [inline Car.java]: Optional: it is possible to use the ternary operator.

File: CarDoor.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package com.mateakademia.homework.homework3.task32.lamborghinicar;
+
+public class CarDoor {
+
+    private boolean IsDoorOpened;
+    private boolean IsWindowOpened;
+
+    CarDoor(){
```
Comment (Anderwerty) [inline CarDoor.java]: Add access modifier.

File: CarDoor.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      IsDoorOpened = false;
```
Comment (Anderwerty) [inline CarDoor.java]: False is the default value for the boolean variable. So lines
```
 IsDoorOpened = false;
 IsWindowOpened = false;
```
are redundent.

File: CarDoor.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
+package com.mateakademia.homework.homework3.task32.lamborghinicar;
+
+public class CarDoor {
+
+    private boolean IsDoorOpened;
```
Comment (Anderwerty) [inline CarDoor.java]: https://mate-academy.github.io/style-guides/java/java.html#s5.2.5-non-constant-field-names

File: CarDoor.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    private boolean IsWindowOpened;
```
Comment (Anderwerty) [inline CarDoor.java]: The same

File: CarDoor.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public boolean getDoor () {
+        return IsDoorOpened;
+    }
+
+    public boolean getWindow () {
+        return IsWindowOpened;
+    }
+
```
Comment (Anderwerty) [inline CarDoor.java:50]: A redundant blank line should be removed.

File: CarDoor.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+        public void showInfo(){
+            System.out.println("The car door  is " + ((getDoor()?"opened":"closed")));
+            System.out.println("The car window  is " + ((getWindow()?"opened":"closed")));
+        }
+
```
Comment (Anderwerty) [inline CarDoor.java]: A redundant blank line. Remove it

File: CarDoor.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+}
+
```
Comment (Anderwerty) [inline CarDoor.java:61]: Redundant blank lines. Remove them.

File: CarWheel.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+        public CarWheel(){
+            tireTag = 1.0;
+        }
+
+        public CarWheel(double tireTag){
+            if (checkTire(tireTag)){
+                this.tireTag = tireTag;
+            } else {
+                System.out.println("It's cann't  create CarWheel");
```
Comment (Anderwerty) [inline CarWheel.java:17]: You will create a new object but tireTag will be 0.

File: CarWheel.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+            }
+        }
+
+        // method  check  value inside of [0,1]
+        private boolean checkTire(double tireTag) {
```
Comment (Anderwerty) [inline CarWheel.java]: It seems this method has two responsibility check if value inside of [0,1] and print information.
Let's try to use SOLID principle (S- single responsibility)

File: CarWheel.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
+package com.mateakademia.homework.homework3.task32.lamborghinicar;
+
+    public class CarWheel {
+
+        private double tireTag;
```
Comment (Anderwerty) [inline CarWheel.java]: Order:
Constants, fields,...

**Review by Anderwerty — CHANGES_REQUESTED** [review]
Please resolve comments

File: CarWheel.java [hunk 14]
```diff
@@ -19,15 +19,10 @@ public CarWheel(double tireTag){
...
-
-            if (!chkRes) {
-                System.out.println(" Wrong number tire wear, diapason is [0,1].");
-            } else {
-                System.out.println("It's right number tire wear, diapason is [0,1].");
-            }
-            return chkRes;
+        private boolean checkTire (double tireTag) {
+            boolean result = ((tireTag >= TIRE_OLD) && (tireTag <= TIRE_NEW));
+            if (!result) System.out.println(" Wrong number tire wear, diapason is [0,1].");
```
Comment (Anderwerty) [inline CarWheel.java]: https://mate-academy.github.io/style-guides/java/java.html#s4.1.1-braces-always-used
Braces are used with if, else, for, do and while statements, even when the body is empty or contains only a single statement.

**Review by Anderwerty — CHANGES_REQUESTED** [review]

**Review by Anderwerty — APPROVED** [review]

**Pull request merged**